core/meta/inc/TClass.h

```diff
@@ -132,6 +132,26 @@ friend class TStreamerInfo;
                         // in ROOT Meta w/o interpreter information
    };
 
+   // "Typed" pointer that recalls how TClass::New allocated the object.
+   // It is returned by TClass:NewObject and should be passed to TClass::DeleteArray or TClass::Destructor
+   // to delete the object.
+   // It is also used in TVirtualCollectionProxy for the same reasons.
+   class ObjectPtr
```

Comment:
Should this be a `struct` such that aggregate init can be used (no ctors), and all getters can vanish except for `operator bool`?